dna/models/lstm_model.py

```diff
@@ -0,0 +1,119 @@
+import torch
+import numpy as np
+
+from .base_models import PyTorchRegressionRankSubsetModelBase
+from .base_models import PyTorchModelBase
+from .torch_modules.lstm_mlp import LSTMMLP
+from dna.data import RNNDataLoader
+
+
+class LSTMModel(PyTorchRegressionRankSubsetModelBase):
+    def __init__(
+            self, activation_name: str, hidden_state_size: int, lstm_n_layers: int, dropout: float,
+            output_n_hidden_layers: int, output_hidden_layer_size: int, use_batch_norm: bool, use_skip: bool = False, *,
+            device: str = 'cuda:0', seed: int = 0
+    ):
+        super().__init__(y_dtype=torch.float32, seed=seed, device=device)
+
+        self.activation_name = activation_name
+        self.hidden_state_size = hidden_state_size
+        self.lstm_n_layers = lstm_n_layers
+        self.dropout = dropout
+        self.output_n_hidden_layers = output_n_hidden_layers
+        self.output_hidden_layer_size = output_hidden_layer_size
+        self.use_batch_norm = use_batch_norm
+        self.use_skip = use_skip
+        self.device = device
+        self.seed = seed
+        self._data_loader_seed = seed + 1
+        self._model_seed = seed + 2
+
+        self.pipeline_structures = None
+        self.num_primitives = None
+        self.primitive_name_to_enc = None
+        self.target_key = 'test_f1_macro'
+        self.batch_group_key = 'pipeline_structure'
+        self.pipeline_key = 'pipeline'
+        self.steps_key = 'steps'
+        self.prim_name_key = 'name'
+        self.prim_inputs_key = 'inputs'
+        self.features_key = 'metafeatures'
+
+    def fit(self, train_data, n_epochs, learning_rate, batch_size, drop_last, *, validation_data=None, output_dir=None,
+            verbose=False):
+
+        # Get the mapping of primitives to their one hot encoding
+        self.primitive_name_to_enc = self._get_primitive_name_to_enc(train_data=train_data)
+
+        PyTorchModelBase.fit(
+            self, train_data, n_epochs, learning_rate, batch_size, drop_last, validation_data=validation_data,
+            output_dir=output_dir, verbose=verbose
+        )
+
+    def _get_primitive_name_to_enc(self, train_data):
```

Comment:
Code like this is starting to show up in a lot of places. We need to consolidate and reuse it.

Reply:
I see now that you moved this code.